### ISSUE TITLE
Add returns_with_once

### ISSUE THREAD

**Issue by ryo33** [ryo33/mry#5]

**Comment by thedodd** [comment]
Yes, this is good. Currently with multiple calls, taking place at different times and in different order, mry doesn't quite seem to handle these cases correctly.

**Comment by ryo33** [comment]
@thedodd Thank you for your comment! Could you let me know some details of your use case?
